src/contexts/events.tsx

```diff
@@ -122,6 +140,14 @@ const useEvents = () => {
   const previousChainId = useRef<number | undefined>(undefined);
   const userClickedState = useRef<boolean>(false);
 
+  const [_scheduledQuotes, setScheduledQuotes] = useState<
```

Comment:
Since we never read `_scheduledQuotes` directly but only inside the `setScheduledQuotes` update function, we could alternatively just make this a React ref. That way no rerender would be triggered when `scheduleQuote` is executed as no state is changed.

Reply:
Refactored to use a ref instead.